extra_data/reader.py

```diff
@@ -1642,7 +1644,26 @@ def train_timestamps(self, labelled=False):
         arr[arr == epoch] = 'NaT'  # Not a Time
         if labelled:
             import pandas as pd
-            return pd.Series(arr, index=self.train_ids)
+            series = pd.Series(arr, index=self.train_ids).dt.tz_localize('UTC')
+            return series.dt.tz_convert('Europe/Berlin') if euxfel_local_time else series
+        elif pydatetime:
+            from datetime import datetime, timezone
+            res = []
+            for npdt in arr:
+                pydt = npdt.astype('datetime64[ms]').item()
+                if pydt is not None:  # Numpy NaT becomes None
+                    pydt = pydt.replace(tzinfo=timezone.utc)
+                    if euxfel_local_time:
+                        from zoneinfo import ZoneInfo
+                        pydt = pydt.astimezone(ZoneInfo('Europe/Berlin'))
+                res.append(pydt)
+            return res
+        elif euxfel_local_time:
+            raise ValueError(
+                'The euxfel_local_time option '
+                + 'can only be invoked if either labelled or pydatetime '
```

Comment:
Nitpick: you invoke (or call) a method, not a parameter.

```suggestion
                + 'can only be used if either labelled or pydatetime '
```